homeassistant/components/apcupsd/__init__.py

```diff
@@ -1,60 +1,130 @@
 """Support for APCUPSd via its Network Information Server (NIS)."""
-from datetime import timedelta
+from __future__ import annotations
+
 import logging
+from datetime import timedelta
+from typing import Final
 
-from apcaccess import status
 import voluptuous as vol
+from apcaccess import status
 
-from homeassistant.const import CONF_HOST, CONF_PORT
-from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
 from homeassistant.helpers.typing import ConfigType
 from homeassistant.util import Throttle
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_HOST = "localhost"
-DEFAULT_PORT = 3551
-DOMAIN = "apcupsd"
-
-KEY_STATUS = "STATFLAG"
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+DOMAIN: Final = "apcupsd"
+KEY_STATUS: Final = "STATFLAG"
+VALUE_ONLINE: Final = 8
+PLATFORMS: Final = (Platform.SENSOR, Platform.BINARY_SENSOR)
+MIN_TIME_BETWEEN_UPDATES: Final = timedelta(seconds=60)
 
-VALUE_ONLINE = 8
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
-                vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-                vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+                vol.Optional(CONF_HOST, default="localhost"): cv.string,
+                vol.Optional(CONF_PORT, default=3551): cv.port,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass: HomeAssistant, config: ConfigType) -> bool:
-    """Use config values to set up a function enabling status retrieval."""
-    conf = config[DOMAIN]
-    host = conf[CONF_HOST]
-    port = conf[CONF_PORT]
+def _async_get_imported_entry(hass: HomeAssistant) -> ConfigEntry | None:
+    """Return the config entry imported from YAML configurations, None is returned if
+    not found."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.source == SOURCE_IMPORT:
+            return entry
+    return None
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up integration from legacy YAML configurations."""
+    if (conf := config.get(DOMAIN)) is None:
+        return True
+
+    _LOGGER.warning(
```

Comment:
We only need one warning when importing config YAML. Please consolidate this warning with the warning in the sensor platform.

Reply:
Done!